### PR TITLE
cleanup: use patches instead to patchesStrategicMerge for kustomize.

### PR DESCRIPTION
**Pull request by liangyuanpeng**
**What type of PR is this?**

/kind cleanup
**What this PR does / why we need it**:

patchesStrategicMerge is deprecated, following the warning message to run command of `kustomize edit fix` for update it.

```shell
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: charts/karmada/_crds/kustomization.yaml [hunk 1]
```diff
@@ -18,6 +18,8 @@ resources:
 - bases/remedy/remedy.karmada.io_remedies.yaml
 - bases/apps/apps.karmada.io_workloadrebalancers.yaml
 
-patchesStrategicMerge:
-- patches/webhook_in_resourcebindings.yaml
-- patches/webhook_in_clusterresourcebindings.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (XiShanYongYe-Chang) [inline charts/karmada/_crds/kustomization.yaml:22]: Is it okay if these two lines are not placed at the top of the file?

Comment (liangyuanpeng) [reply charts/karmada/_crds/kustomization.yaml]: it's generate by kustomize, i will suggesstion to keep it for consistent with kustomize.

now i Removed it for resolve comment.

Comment (XiShanYongYe-Chang) [reply charts/karmada/_crds/kustomization.yaml]: Aren't these two lines necessary?

Comment (XiShanYongYe-Chang) [reply charts/karmada/_crds/kustomization.yaml]: Thanks, I get it. 

/lgtm
/approve

Comment (liangyuanpeng) [reply charts/karmada/_crds/kustomization.yaml]: Thanks you!

**Pull request merged**